skdownscale/pointwise_models/gard.py

```diff
@@ -232,14 +284,87 @@ def predict(self, X):
             # for mean/weight cases, this fills nans when all analogs
             # were below thresh
             predicted = np.nan_to_num(predicted, nan=0.0)
-
-        if self.stats:
-            # calculate the standard deviation of the anlogs
-            if self.thresh is None:
-                self.stats_['error'] = analogs.std(axis=1)
-            else:
-                self.stats_['error'] = analogs.where(analog_mask).std(axis=1)
-                # calculate the probability of precip
-                self.stats_['pop'] = np.where(analog_mask, 1, 0).mean(axis=1)
+            self.prediction_error_ = masked_analogs.std(axis=1)
+            self.exceedance_prob_ = np.where(analog_mask, 1, 0).mean(axis=1)
+        else:
+            self.prediction_error_ = analogs.std(axis=1)
+            self.exceedance_prob_ = np.ones(len(X), dtype=np.float64)
 
         return predicted
+
+
+class PureRegression(RegressorMixin, BaseEstimator):
+    """ PureRegression
+    Parameters
+    ----------
+    thresh : float
+        Subset analogs based on threshold
+    logistic_kwargs : dict
+        Dictionary of keyword arguments to pass to logistic regression model
+    linear_kwargs : dict
+        Dictionary of keyword arguments to pass to linear regression model
+
+    Attributes
+    ----------
+    kdtree_ : sklearn.neighbors.KDTree
+        KDTree object
+
+    Notes
+    -----
+    GARD pure regression models store prediction error  (`model.prediction_error_`) for the last fit and
+    exceedance probabilities (`model.exceedance_prob_`) for the last prediction.
+    """
+
+    _fit_attributes = [
+        'stats_',
+        'logistic_model_',
+        'linear_model_',
+        'prediction_error_',
+        'exceedance_prob_',
+    ]
+
+    def __init__(
+        self, thresh=None, logistic_kwargs=None, linear_kwargs=None,
+    ):
+        self.thresh = thresh
+        self.logistic_kwargs = logistic_kwargs
+        self.linear_kwargs = linear_kwargs
+
+    def fit(self, X, y):
+        X, y = self._validate_data(X, y=y, y_numeric=True)
+        self.stats_ = {}
+
+        if self.thresh is not None:
+            exceed_ind = y > self.thresh
+            binary_y = exceed_ind.astype(np.int8)
+            logistic_kwargs = default_none_kwargs(self.logistic_kwargs)
+            self.logistic_model_ = LogisticRegression(**logistic_kwargs).fit(X, binary_y)
+        else:
+            exceed_ind = np.ones(len(y), dtype=bool)
+
+        linear_kwargs = default_none_kwargs(self.linear_kwargs)
+        self.linear_model_ = LinearRegression(**linear_kwargs).fit(X[exceed_ind], y[exceed_ind])
+
+        y_hat = self.linear_model_.predict(X[exceed_ind])
+        error = mean_squared_error(y[exceed_ind], y_hat, squared=False)
+        self.prediction_error_ = np.full(shape=len(X), fill_value=error)
+        self.exceedance_prob_ = []  # populated in predict
```

Comment:
Wont this lead to inconsistent lengths in `prediction_error_` and `exceedance_prob_`? That seems like a confusing outcome.

Reply:
Good catch! Modified to save the error value only in fit, and construct both `prediction_error_` and `exceedance_prob_` in predict.